src/core/mac/mac.hpp

```diff
@@ -751,6 +751,22 @@ class Mac : public InstanceLocator, private NonCopyable
      */
     Error GetRegion(uint16_t &aRegionCode) const;
 
+    /**
+     * Gets the Wake-up channel.
+     *
+     * @returns Wake-up channel.
+     *
+     */
+    uint8_t GetWakeupChannel(void) const { return mWakeupChannel; }
+
+    /**
+     * Sets the Wake-up channel.
+     *
+     * @param[in]  aChannel  The Wake-up channel.
+     *
+     */
+    void SetWakeupChannel(uint8_t aChannel);
```

Comment:
We could consider adding `#if` guards around the methods/functions that set or get `WakeupChannel` in `Mac` and/or `LinkRaw`.  I know there are config are defined https://github.com/openthread/openthread/pull/10728 (`OPENTHREAD_CONFIG_WAKEUP_COORDINATOR_ENABLE` and `OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE`).


My thinking is that if a device doesn't support this function, it won't need those methods, and providing those may cause confusion. I assume such a device would still need to parse and understand the dataset TLV, but it might not need to set or get `WakeupChannel` on the MAC layer.

Thoughts?

Reply:
Agree. What if we add a follow-up task to https://github.com/openthread/openthread/issues/10725 for now? To wait for https://github.com/openthread/openthread/pull/10728 to be merged with the needed changes.

Reply:
Either way can work. Thanks :)

Reply:
Done.